components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -963,82 +1036,86 @@ static void psa_key_management_operation(void)
                             break;
                         }
 
-
                         bytes_read = psa_read(msg.handle, 2,
                                               parameter, parameter_size);
-
                         if (bytes_read != parameter_size) {
                             SPM_PANIC("SPM read length mismatch");
                         }
                     }
+
                     status = psa_generate_key(psa_key_mng.key,
                                               psa_key_mng.type,
                                               bits,
                                               parameter, parameter_size);
                     mbedtls_free(parameter);
-
                     break;
                 }
+
                 default: {
                     status = PSA_ERROR_NOT_SUPPORTED;
                     break;
                 }
             }
+
             break;
         }
-        case PSA_IPC_DISCONNECT: {
-            break; /* do nothing*/
-        }
+
         default: {
-            status = PSA_ERROR_NOT_SUPPORTED;
+            SPM_PANIC("Unexpected message type %d!", (int)(msg.type));
             break;
         }
     }
+
     psa_reply(msg.handle, status);
 }
 
 static void psa_entropy_operation(void)
 {
     psa_msg_t msg = { 0 };
     psa_status_t status = PSA_SUCCESS;
-    psa_get(PSA_ENTROPY_INJECT, &msg);
 
-#if ( defined(MBEDTLS_ENTROPY_NV_SEED) && defined(MBEDTLS_PSA_HAS_ITS_IO) )
+    psa_get(PSA_ENTROPY_INJECT, &msg);
     switch (msg.type) {
-        case PSA_IPC_CONNECT: {
-            break; /* do nothing */
+        case PSA_IPC_CONNECT:
+        case PSA_IPC_DISCONNECT: {
+            status = PSA_SUCCESS;
+            break;
         }
+
         case PSA_IPC_CALL: {
+#if (defined(MBEDTLS_ENTROPY_NV_SEED) && defined(MBEDTLS_PSA_HAS_ITS_IO))
```

Comment:
Two `)`?

(In the end of the line)

Reply:
@cmonr i hope the following graphic helps
```
#if (defined(MBEDTLS_ENTROPY_NV_SEED) && defined(MBEDTLS_PSA_HAS_ITS_IO))
    |       |                       |           |                      ||
    |       -------------------------           ------------------------|
    |                                                                   |
    ---------------------------------------------------------------------
```